server/graphql/core/src/loader/program_indicator_value.rs

```diff
@@ -0,0 +1,128 @@
+use actix_web::web::Data;
+use async_graphql::dataloader::*;
+use async_graphql::*;
+use repository::{EqualFilter, RepositoryError};
+use service::service_provider::{ServiceContext, ServiceProvider};
+use std::collections::HashMap;
+
+use super::IdPair;
+
+#[derive(Clone)]
+pub struct IndicatorValuePayload {
+    pub period_id: String,
+    pub supplier_store_id: String,
+    pub customer_name_id: String,
+}
+
+pub type IndicatorValueLoaderInput = IdPair<IndicatorValuePayload>;
+impl IndicatorValueLoaderInput {
+    pub fn new(line_id: &str, column_id: &str, extra_filter: IndicatorValuePayload) -> Self {
+        IndicatorValueLoaderInput {
+            primary_id: line_id.to_string(),
+            secondary_id: column_id.to_string(),
+            // later add store and period and facility ids to this payload
+            payload: extra_filter,
+        }
+    }
+}
+
+pub struct IndicatorValueLoader {
+    pub service_provider: Data<ServiceProvider>,
+}
+
+impl Loader<IndicatorValueLoaderInput> for IndicatorValueLoader {
+    type Value = ProgramIndicatorValue;
+    type Error = async_graphql::Error;
+
+    async fn load(
+        &self,
+        loader_inputs: &[IndicatorValueLoaderInput],
+    ) -> Result<HashMap<IndicatorValueLoaderInput, Self::Value>, Self::Error> {
+        let service_context = self.service_provider.basic_context()?;
+
+        let (period_id, supplier_store_id, customer_name_id) =
+            // TODO replace with logic to not assume only one combination queried at any time.
+            if let Some(loader_input) = loader_inputs.first() {
```

Comment:
This assumes that only one requisition is queried at anytime, otherwise values might get mixed on loader bundling